### PR TITLE
Arrow, Ellipse, Box, Polygon[Group]

### PR DESCRIPTION
**Pull request by floryst**
No :snake: to be found here

### CODE REVIEW

File: Modules/Core/SpatialObjects/include/itkBoxSpatialObject.hxx [hunk 1]
```diff
@@ -105,46 +89,45 @@ BoxSpatialObject< TDimension >
...
-    pnt2[i] = m_PositionInObjectSpace[i] + m_SizeInObjectSpace[i];
-    }
+  m_Position = pos;
 
-  bb->SetMinimum(pnt1);
-  bb->SetMaximum(pnt1);
-  bb->ConsiderPoint(pnt2);
-  bb->ComputeBoundingBox();
+  // Transform the corners of the bounding box
+  this->ComputeMyBoundingBox();
```
Comment (floryst) [inline Modules/Core/SpatialObjects/include/itkBoxSpatialObject.hxx]: computing corners means we need to compute the bounding box..........

File: Modules/Core/SpatialObjects/include/itkBoxSpatialObject.hxx [hunk 2]
```diff
@@ -32,45 +32,29 @@ BoxSpatialObject< TDimension >
...
+::IsInside(const PointType & worldPoint, unsigned int depth,
   const std::string & name) const
 {
-  itkDebugMacro("Checking the point [" << point << "] is in the box");
+  itkDebugMacro("Checking the point [" << worldPoint << "] is in the box");
 
   if( this->GetTypeName().find( name ) != std::string::npos )
     {
-    if( this->GetMyBounds()->IsInside( point ) )
+    if( this->GetMyBounds()->IsInside( worldPoint ) )
```
Comment (aylward) [inline Modules/Core/SpatialObjects/include/itkBoxSpatialObject.hxx]: Change GetMyBounds to GetMyBoundingBox

File: Modules/Core/SpatialObjects/include/itkBoxSpatialObject.hxx [hunk 3]
```diff
@@ -105,29 +89,42 @@ BoxSpatialObject< TDimension >
...
+    {
+    pnt1[i] = m_PositionInObjectSpace[i];
+    pnt2[i] = m_PositionInObjectSpace[i] + m_SizeInObjectSpace[i];
+    }
+
+  pnt1 = this->GetObjectToWorldTransform()->TransformPoint(pnt1);
+  pnt2 = this->GetObjectToWorldTransform()->TransformPoint(pnt2);
 
-  // refresh the bounding box with the transformed corners
   const_cast< BoundingBoxType * >( this->GetMyBounds() )
```
Comment (aylward) [inline Modules/Core/SpatialObjects/include/itkBoxSpatialObject.hxx]: Ditto

File: Modules/Core/SpatialObjects/include/itkEllipseSpatialObject.hxx [hunk 4]
```diff
@@ -158,22 +127,39 @@ EllipseSpatialObject< TDimension >
     }
 
   // refresh the bounding box with the transformed corners
-  const_cast< BoundingBoxType * >( this->GetObjectBounds() )
+  const_cast< BoundingBoxType * >( this->GetMyBounds() )
```
Comment (aylward) [inline Modules/Core/SpatialObjects/include/itkEllipseSpatialObject.hxx]: Ditto

**Pull request merged**